src/GUI/Overlay/OverlaySystem.cpp

```diff
@@ -52,9 +52,19 @@ void OverlaySystem::_update() {
 
     if (context->style.font != nullptr) {
         nk_input_end(context);
-        for (auto &&[name, overlay] : _overlays) {
-            overlay->update(context);
+        if (_isEnabled) {
+            for (auto &&[name, overlay] : _overlays) {
+                overlay->update(context);
+            }
         }
         nk_input_begin(context);
```

Comment:
Shouldn't we just do
```
if (!_isEnabled)
    return;
```
at the start of this function?

Reply:
The `nk_input_begin` and `nk_input_end` needs to be called in order to update the hitboxes properly. For some reason, if I stop calling them, the input is still catched by nuklear even if the windows are hidden. I can probably investigate further but I 100% believe we'll get rid of nuklear anyway in favor of imgui at some point.

Maybe I should add a comment? I agree that it doesn't look clear.

Reply:
Nevermind! I just tested it again and now it works. I think I solved it, without knowning, by changing this line mentioned here: https://github.com/OpenEnroth/OpenEnroth/pull/1625#discussion_r1590176248